sagemaker_processing/spark_distributed_data_processing/sagemaker-spark-processing.ipynb

```diff
@@ -0,0 +1,562 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import logging\n",
+    "import sagemaker\n",
+    "from time import gmtime, strftime\n",
+    "\n",
+    "sagemaker_logger = logging.getLogger(\"sagemaker\")\n",
+    "sagemaker_logger.setLevel(logging.INFO)\n",
+    "sagemaker_logger.addHandler(logging.StreamHandler())\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "bucket = sagemaker_session.default_bucket()\n",
+    "role = sagemaker.get_execution_role()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, you'll download the example dataset from a SageMaker staging bucket."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Fetch the dataset from the SageMaker bucket\n",
+    "!wget https://s3-us-west-2.amazonaws.com/sparkml-mleap/data/abalone/abalone.csv -O ./data/abalone.csv"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Write the PySpark script\n",
+    "\n",
+    "The source for a preprocessing script is in the cell below. The cell uses the `%%writefile` directive to save this file locally. This script does some basic feature engineering on a raw input dataset. In this example, the dataset is the [Abalone Data Set](https://archive.ics.uci.edu/ml/datasets/abalone) and the code below performs string indexing, one hot encoding, vector assembly, and combines them into a pipeline to perform these transformations in order. The script then does an 80-20 split to produce training and validation datasets as output."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile ./code/preprocess.py\n",
+    "from __future__ import print_function\n",
+    "from __future__ import unicode_literals\n",
+    "\n",
+    "import time\n",
+    "import sys\n",
+    "import os\n",
+    "import shutil\n",
+    "import csv\n",
+    "\n",
+    "import pyspark\n",
+    "from pyspark.sql import SparkSession\n",
+    "from pyspark.ml import Pipeline\n",
+    "from pyspark.sql.types import StructField, StructType, StringType, DoubleType\n",
+    "from pyspark.ml.feature import StringIndexer, VectorIndexer, OneHotEncoder, VectorAssembler\n",
+    "from pyspark.sql.functions import *\n",
+    "\n",
+    "\n",
+    "def csv_line(data):\n",
+    "    r = ','.join(str(d) for d in data[1])\n",
+    "    return str(data[0]) + \",\" + r\n",
+    "\n",
+    "\n",
+    "def main():\n",
+    "    spark = SparkSession.builder.appName(\"PySparkApp\").getOrCreate()\n",
+    "    \n",
+    "    # Convert command line args into a map of args\n",
+    "    args_iter = iter(sys.argv[1:])\n",
+    "    args = dict(zip(args_iter, args_iter))\n",
```

Comment:
For consistency, do you want to use `argparse` here as you do in the other script?

Reply:
Updated